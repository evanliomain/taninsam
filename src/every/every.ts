/**
 * @module array=>boolean
 */
/**
 * Determines whether all the members of an array satisfy the specified test.
 * @param iteratee A function that accepts up to three arguments.
 * @return the function to apply on the array to determine
 * if its satisfy the specified test
 * @example
 * ```
 * every<number>(x => x < 6)([1, 2, 3, 4, 5]) // true
 * ```
 * @example Using the chain
 * ```
 * chain([1, 2, 3, 4, 5])
 *   .chain(every<number>(x => x < 6))
 *   .value() // true
 * ```
 */
export function every<T>(
  iteratee: (value: T, index: number, array: ReadonlyArray<T>) => boolean
): (array: ReadonlyArray<T>) => boolean {
  return (array: ReadonlyArray<T>) => array.every(iteratee);
}
