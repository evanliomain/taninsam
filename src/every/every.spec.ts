import { describe, expect, test } from 'vitest';
import { every } from './every';

describe('every function', () => {
  const input = [1, 2, 3, 4, 5];

  test('is a pure function', () => {
    every((x: number): boolean => x < 6)(input);
    expect(input).toMatchSnapshot();
  });

  test('[1, 2, 3, 4, 5] |> every === true', () => {
    expect(every((x: number): boolean => x < 6)(input)).toMatchSnapshot();
  });
});
