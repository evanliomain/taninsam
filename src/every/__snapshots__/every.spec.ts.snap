// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`every function [1, 2, 3, 4, 5] |> every === true 1`] = `true`;

exports[`every function is a pure function 1`] = `
Array [
  1,
  2,
  3,
  4,
  5,
]
`;
